mondrian-generator.js

```diff
@@ -42,69 +35,124 @@ function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min + 1) + min);
 }
 
-function getLinePositions() {
-  const numXLines = getRandomInt(2, 4);
-  const numYLines = getRandomInt(2, 4)
-  x = [];
-  for (i = 0; i < numXLines; i++) {
-    x.push(getRandomInt(10, 380))
-  };
-  y = [];
-  for (i = 0, t = numYLines; i < t; i++) {
-    y.push(getRandomInt(10, 380))
+function orientXYtoTopLeft(xInnerLines, yInnerLines) {
+  const xLineStarts = xInnerLines.sort((a, b) => a - b);
+  const yLineStarts = yInnerLines.sort((a, b) => a - b).reverse();
+
+  return { xLineStarts, yLineStarts }
+}
+
+function addBorders(innerLines) {
+  const BORDER_WIDTH = 2
+  const linesWithBorders = [BORDER_WIDTH, ...innerLines, CANVAS_WIDTH - BORDER_WIDTH]
+  return linesWithBorders
+}
+
+function getInnerLines(numLines) {
+  const innerLines = [];
+  const PADDING_MARGIN = 10
+
+  for (let i = 0; i < numLines; i++) {
+    innerLines.push(getRandomInt(PADDING_MARGIN, CANVAS_WIDTH - PADDING_MARGIN))
   };
-  x = x.sort();
-  y = y.sort().reverse(); // easier to think of 0,0 as top left
+  const linesWithBorders = addBorders(innerLines)
+  return linesWithBorders
+}
 
-  // add border
-  x.unshift(2);
-  x.push(398);
-  y.unshift(2);
-  y.push(398);
+function getLineStarts() {
+  const numXLines = getRandomInt(2, 4);
+  const numYLines = getRandomInt(2, 4)
+  const xInnerLines = getInnerLines(numXLines)
+  const yInnerLines = getInnerLines(numYLines)
+  const { xLineStarts, yLineStarts } = orientXYtoTopLeft(xInnerLines, yInnerLines)
 
-  return { x, y }
+  return { xLineStarts, yLineStarts }
 }
 
+const CANVAS_WIDTH = 400
+const CANVAS_HEIGHT = 400
```

Comment:
Setting the width and height of the `<canvas/>` in just a single place here (and not in the index.html) is really important. It will allow the dimensions of the canvas to be set separately from other variables that rely on the canvas dimensions, like `PADDING_MARGIN` or `BORDER_WIDTH`.